### ISSUE TITLE
Memory leak detected by LeakCanary

### ISSUE THREAD

**Issue by GersomR-afk** [hCaptcha/hcaptcha-android-sdk#122]
---

Hello,

I'm using the hCaptcha SDK in my app and I'm experiencing a memory leak issue. Upon finishing the captcha, a memory leak occurs.

I've attached my implementation code and the LeakCanary log below. Let me know if you need any more information and whether there are any workarounds or fixes available to address this issue.

Implementation code:
I'm using Jetpack Compose, with an activity extending FragmentActivity. Then, from inside a viewmodel, I have a function like this:

```
fun signInViaEmail(activity: FragmentActivity, onComplete: () -> Unit) {
      val hCaptcha = HCaptcha.getClient(activity)
      hCaptcha.addOnSuccessListener { response -> 
            // Check the response
            onComplete()
      }
}
```
This function is then in turn called from my Jetpack Compose UI where LocalContext.current is being casted to FragmentActivity.

LeakCanary log:
```
┬───
│ GC Root: System class
│
├─ android.app.ActivityThread class
│    Leaking: NO (MessageQueue↓ is not leaking and a class is never leaking)
│    ↓ static ActivityThread.sMainThreadHandler
├─ android.app.ActivityThread$H instance
│    Leaking: NO (MessageQueue↓ is not leaking)
│    ↓ Handler.mQueue
├─ android.os.MessageQueue instance
│    Leaking: NO (MessageQueue#mQuitting is false)
│    HandlerThread: "main"
│    ↓ MessageQueue[3]
│                  ~~~
├─ android.os.Message instance
│    Leaking: UNKNOWN
│    Retaining 63.0 kB in 1481 objects
│    Message.what = 0
│    Message.when = 396137 (74325 ms after heap dump)
│    Message.obj = null
│    Message.callback = instance @328107720 of com.hcaptcha.sdk.tasks.Task$1
│    Message.target = instance @325042816 of android.os.Handler
│    ↓ Message.callback
│              ~~~~~~~~
├─ com.hcaptcha.sdk.tasks.Task$1 instance
│    Leaking: UNKNOWN
│    Retaining 62.9 kB in 1480 objects
│    Anonymous class implementing java.lang.Runnable
│    ↓ Task$1.this$0
│             ~~~~~~
├─ com.hcaptcha.sdk.HCaptcha instance
│    Leaking: UNKNOWN
│    Retaining 62.9 kB in 1479 objects
│    activity instance of REDACTED.MainActivity with mDestroyed = false
│    ↓ HCaptcha.captchaVerifier
│               ~~~~~~~~~~~~~~~
╰→ com.hcaptcha.sdk.HCaptchaDialogFragment instance
​     Leaking: YES (ObjectWatcher was watching this because com.hcaptcha.sdk.HCaptchaDialogFragment received
​     Fragment#onDestroy() callback and Fragment#mFragmentManager is null)
​     Retaining 62.5 kB in 1466 objects
​     key = ae934461-4e99-4218-b07e-88cf2948fb5c
​     watchDurationMillis = 45527
​     retainedDurationMillis = 40526

METADATA

Build.VERSION.SDK_INT: 32
Build.MANUFACTURER: Google
LeakCanary version: 2.10
App process name: REDACTED
Class count: 36680
Instance count: 241264
Primitive array count: 161447
Object array count: 37254
Thread count: 130
Heap total bytes: 37370546
Bitmap count: 13
Bitmap total bytes: 1497189
Large bitmap count: 0
Large bitmap total bytes: 0
Db 1: open /data/user/0/REDACTED/databases/notification_db
Db 2: closed /data/user/0/REDACTED/databases/google_app_measurement_local.db
Db 3: open /data/user/0/REDACTED/databases/app_db
Db 4: open /data/user/0/REDACTED/databases/com.google.android.datatransport.events
Db 5: open /data/user/0/REDACTED/no_backup/androidx.work.workdb
Stats: LruCache[maxSize=3000,hits=126963,misses=237397,hitRate=34%]
RandomAccess[bytes=11735220,reads=237397,travel=107509458964,range=40473978,size=52966537]
Analysis duration: 21588 ms
```

Thank you in advance for your help!

--- 

I hope this helps! Let me know if you have any other questions.

**Comment by CAMOBAP** [comment]
Thanks for reporting this. This is expected behavior. We cache the fragment instance inside SDK to speed up the next `HCaptcha.verifyWithHCaptcha` calls.

Once you are done with verification, you can call `HCaptcha.reset()` to release all allocated resources including the strong reference to `com.hcaptcha.sdk.HCaptchaDialogFragment`

`HCaptcha.reset()` available since `3.8.0` version

@GersomR-afk any feedback is appreciated, please let us know if proposed solution works for you

**Comment by GersomR-afk** [comment]
Hello @CAMOBAP! Thank you for getting back to me so quickly and providing a solution to the memory leak issue. I apologize for not mentioning earlier that I wasn't using the ``3.8.0`` version of the hCaptcha SDK.

I really appreciate your help and the suggested solution of calling ``HCaptcha.reset()`` to release all allocated resources. It's great to know that this function is available in the 3.8.0 version.

However, I have a suggestion for improvement. It would be beneficial if this information about using ``HCaptcha.reset()`` to prevent memory leaks were made more explicit in the documentation, particularly in the README file. Clarifying this point would help developers like me who might overlook it and face similar memory leak issues.

Once again, thank you for your prompt assistance and the recommended solution. The issue's been resolved now. If I encounter any further issues or have additional feedback, I'll be sure to let you know.

**Comment by CAMOBAP** [comment]
@GersomR-afk thanks for the feedback, we add this information to README to improve our documentation
